### ISSUE TITLE
Load torchscript failed: PytorchStreamReader failed locating file constants.pkl: file not found

### ISSUE THREAD

**Issue by 408550969** [pnnx/pnnx#50]
E:\yolov5-face-master\fisheye>E:\pnnx-20220720-windows\pnnx.exe E:\yolov5-face-master\fisheye\yolov5n-widerface-640.pt inputshape=[1,3,224,224]
pnnxparam = E:\yolov5-face-master\fisheye\yolov5n-widerface-640.pnnx.param
pnnxbin = E:\yolov5-face-master\fisheye\yolov5n-widerface-640.pnnx.bin
pnnxpy = E:\yolov5-face-master\fisheye\yolov5n-widerface-640_pnnx.py
ncnnparam = E:\yolov5-face-master\fisheye\yolov5n-widerface-640.ncnn.param
ncnnbin = E:\yolov5-face-master\fisheye\yolov5n-widerface-640.ncnn.bin
ncnnpy = E:\yolov5-face-master\fisheye\yolov5n-widerface-640_ncnn.py
optlevel = 2
device = cpu
inputshape = [1,3,224,224]f32
inputshape2 =
customop =
moduleop =
Load torchscript failed: PytorchStreamReader failed locating file constants.pkl: file not found
Exception raised from valid at ..\..\caffe2\serialize\inline_container.cc:158 (most recent call first):
00007FFA626CA29200007FFA626CA230 c10.dll!c10::Error::Error [<unknown file> @ <unknown line number>]
00007FFA626C9D1E00007FFA626C9CD0 c10.dll!c10::detail::torchCheckFail [<unknown file> @ <unknown line number>]
00007FFA024508A300007FFA02450810 torch_cpu.dll!caffe2::serialize::PyTorchStreamReader::valid [<unknown file> @ <unknown line number>]
00007FFA0244F4AD00007FFA0244F3F0 torch_cpu.dll!caffe2::serialize::PyTorchStreamReader::getRecordID [<unknown file> @ <unknown line number>]
00007FFA0244F30200007FFA0244F2B0 torch_cpu.dll!caffe2::serialize::PyTorchStreamReader::getRecord [<unknown file> @ <unknown line number>]
00007FFA0367F68600007FFA0367F490 torch_cpu.dll!torch::jit::readArchiveAndTensors [<unknown file> @ <unknown line number>]
00007FFA0367DC1F00007FFA0367D4C0 torch_cpu.dll!torch::jit::load [<unknown file> @ <unknown line number>]
00007FFA0367B98900007FFA03679400 torch_cpu.dll!torch::jit::CallStackDebugInfoUnpickler::unpickle [<unknown file> @ <unknown line number>]
00007FFA0367D6CB00007FFA0367D4C0 torch_cpu.dll!torch::jit::load [<unknown file> @ <unknown line number>]
00007FFA0367D16300007FFA0367D040 torch_cpu.dll!torch::jit::load [<unknown file> @ <unknown line number>]
00007FF60D0D6DFE00007FF60D0D5420 pnnx.exe!c10::ivalue::Future::waitAndThrow [<unknown file> @ <unknown line number>]
00007FF60D20347C00007FF60D0D5420 pnnx.exe!c10::ivalue::Future::waitAndThrow [<unknown file> @ <unknown line number>]
00007FFA87BD703400007FFA87BD7020 KERNEL32.DLL!BaseThreadInitThunk [<unknown file> @ <unknown line number>]
00007FFA87D8265100007FFA87D82630 ntdll.dll!RtlUserThreadStart [<unknown file> @ <unknown line number>]

Please export model to torchscript as follows
------------------------------------------
import torch
import torchvision.models as models

net = models.resnet18(pretrained=True)
net = net.eval()

x = torch.rand(1, 3, 224, 224)
mod = torch.jit.trace(net, x)
mod.save("resnet18.pt")
------------------------------------------
你好，我今天转的时候报这个错，可能是由什么原因导致的？我下的版本是这个
[pnnx-20220720-windows.zip](https://github.com/pnnx/pnnx/releases/download/20220720/pnnx-20220720-windows.zip)
我以前用pnnx转过，没有报错，这次不知道是啥情况

**Comment by nihui** [comment]
导出torchscript所使用的pytorch版本是？

**Comment by Phuong38** [comment]
Hello! I have the same problem, did you find the solution?

**Comment by glenn-jocher** [comment]
@408550969 @nihui great news 😃! Your original issue may now be resolved ✅ in Ultraltyics PR https://github.com/ultralytics/ultralytics/pull/3529 which is part of `ultralytics==8.0.129`. This release adds official support for Ultralytics YOLOv8 NCNN export! You can export with CLI:
```
yolo export model=yolov8n.pt format=ncnn
```

or Python:
```python
from ultralytics import YOLO

# Create a model
model = YOLO('yolov8n.pt')

# Export the model to NCNN with arguments
model.export(format='ncnn', half=True, imgsz=640)
```

Output is a `yolov8n_ncnn_model/` directory containing `model.bin`, `model.param` and `metadata.yaml`, along with extra PNNX files. For details see https://github.com/pnnx/pnnx


To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
